src/features/world/scenes/FactionHouseScene.ts

```diff
@@ -3,12 +3,63 @@ import { interactableModalManager } from "../ui/InteractableModals";
 import { translate } from "lib/i18n/translate";
 import { getFactionPrize } from "../ui/factions/weeklyPrize/FactionWeeklyPrize";
 import { Coordinates } from "features/game/expansion/components/MapPlacement";
+import { getFactionWeek } from "features/game/lib/factions";
+import { CollectivePet } from "features/game/types/game";
+import { PET_SLEEP_DURATION } from "../ui/factions/FactionPetPanel";
+import { getFactionPetUpdate } from "../ui/factions/actions/getFactionPetUpdate";
 
 export abstract class FactionHouseScene extends BaseScene {
+  public collectivePet: CollectivePet | undefined;
+  private fetchInterval: NodeJS.Timeout | null = null;
+
   preload() {
     super.preload();
 
     this.load.image("basic_chest", "world/basic_chest.png");
+    this.makeFetchRequest();
+  }
+
+  create() {
+    super.create();
+    this.fetchInterval = setInterval(() => this.makeFetchRequest(), 10 * 1000);
+  }
+
+  async makeFetchRequest() {
+    const { farmId } = this.gameService.state.context;
+    const data = await getFactionPetUpdate({ farmId });
+
+    this.collectivePet = data;
+  }
+
+  shutdown() {
+    if (this.fetchInterval) {
+      clearInterval(this.fetchInterval);
+    }
+  }
+
+  destroy() {
+    if (this.fetchInterval) {
+      clearInterval(this.fetchInterval);
+    }
+  }
+
+  getPetState() {
+    const week = getFactionWeek({ date: new Date() });
+    const beginningOfWeek = new Date(week).getTime();
+    const firstWeek = "2024-07-08";
+
+    if (!this.collectivePet || week === firstWeek) return "pet_hungry";
```

Comment:
The pet is "hungry" for the entire first week no matter how much it's fed?

Reply:
Yep, the pet can be overfed